DEVELOPER_GUIDE.md

```diff
@@ -494,3 +497,15 @@ Pass a list of files or directories to limit your search.
 The Github workflow in [lucene-snapshots.yml](.github/workflows/lucene-snapshots.yml) is a Github worfklow executable by maintainers to build a top-down snapshot build of lucene.
 These snapshots are available to test compatibility with upcoming changes to Lucene by updating the version at [version.properties](buildsrc/version.properties) with the `version-snapshot-sha` version.
 Example: `lucene = 10.0.0-snapshot-2e941fc`.
+
+# Creating uber-JAR/fat-JAR of a module
+An uber-JAR is the JAR, which contains classes from all the libraries, on which your project depends and, of course, the classes of current project.
+
+There might be cases where the developer would like to add some custom logic to the code of a module (like the rest client) and generate anuber-JAR that can be directly used by the dependency management tool.
+
+To do so you will be using [Gradle Shadow plugin](https://imperceptiblethoughts.com/shadow/).
+Go to the `build.gradle` file of the module for which you want to create the uber-JAR.
+Add the shadow plugin:
+```
+apply plugin: 'com.github.johnrengelman.shadow'
+```
```

Comment:
And then what? What do I type now to generate the fat JAR and what does it produce? Feel free to include console output, just walk me through it like a 2yo.

Reply:
Apologies, I had previously added the following steps but somehow they did not make it to the commit. updating the PR

Reply:
This is the console output:
```
(base) garprana@88665a37c3c3 rest-high-level % ../../gradlew shadowJar

> Configure project :qa:os
Cannot add task 'destructiveDistroTest.docker' as a task with that name already exists.
=======================================
OpenSearch Build Hamster says Hello!
  Gradle Version        : 6.6.1
  OS Info               : Mac OS X 12.4 (x86_64)
  JDK Version           : 11 (JDK)
  JAVA_HOME             : /Library/Java/JavaVirtualMachines/amazon-corretto-11.jdk/Contents/Home
  Random Testing Seed   : D0C50CCED424B21E
  In FIPS 140 mode      : false
=======================================

BUILD SUCCESSFUL in 9s
36 actionable tasks: 1 executed, 35 up-to-date

```

Which is not providing a lot of information so have not included it in the document.